core/services/src/main/java/org/exoplatform/services/cms/clouddrives/CloudDriveServiceImpl.java

```diff
@@ -580,4 +588,31 @@ protected void loadConnected(ManageableRepository jcrRepository) {
       sessionProviders.setSessionProvider(null, spOrig);
     }
   }
+
+  @Override
+  public void disconnectCloudDrive(String workspace, String userEmail, String providerId) {
+    long userIdentityId = getCurrentUserIdentityId(identityManager);
+    String currentUserName = ConversationState.getCurrent().getIdentity().getUserId();
+    SessionProvider sessionProvider = sessionProviders.getSessionProvider(null);
+    try {
+      Session userSession = sessionProvider.getSession(workspace, jcrService.getCurrentRepository());
+      String queryString = "select * from " + JCRLocalCloudDrive.ECD_CLOUDDRIVE + " where ecd:localUserName='" + currentUserName
+          + "' AND ecd:provider='" + providerId + "' AND ecd:userEmail='" + userEmail + "'";
+      Query query = userSession.getWorkspace().getQueryManager().createQuery(queryString, Query.SQL);
+      NodeIterator nodeIterator = query.execute().getNodes();
+      if (nodeIterator.hasNext()) {
+        Node cloudDriveNode = nodeIterator.nextNode();
+        cloudDriveNode.remove();
+        userSession.save();
+      }
+      cloudDriveUserSettingsService.deleteCloudDriveUserSettings(userIdentityId);
```

Comment:
you should update the cloud drive settings by deleting only the pair providerId, email